raft-engine: add the `spill-dir` config parameter (#18981)
pingcap/docs: tikv-configuration-file.md

+ If there are multiple disks on your machine, it is recommended to store the data of Raft Engine on a different disk to improve TiKV performance.
+ Default value: `""`

### `spill-dir` <span class="version-mark">New in v8.4.0</span>

> + The auxiliary directory for storing raft log files. When the disk where the `dir` specified is full, new raft logs will be stored under this directory. If this directory does not exist after configuration, it will be created when TiKV is started.
> + If this configuration is not set, it means that the auxiliary directory is disabled.

Reviewer: ```suggestion
+ The auxiliary directory for storing Raft log files. When the disk for the `dir` directory is full, new Raft logs will be stored under this directory. If this auxiliary directory does not exist after configuration, it will be automatically created when TiKV is started.
+ If this configuration is not set, the auxiliary directory is not enabled.
```

> **Note:**
>

> > - This configuration only takes effect when the `dir` and `spill-dir` of the Raft Engine are set to different disk drives.

Reviewer: ```suggestion
> - This configuration takes effect only when the `dir` and `spill-dir` of the Raft Engine are set to different disk drives.
```

> > - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV:

Reviewer 2: ```suggestion
> - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV; otherwise, TiKV won't be able to start:
```

> + The auxiliary directory for storing raft log files. When the disk where the `dir` specified is full, new raft logs will be stored under this directory. If this directory does not exist after configuration, it will be created when TiKV is started.
> + If this configuration is not set, it means that the directory does not exist.
> 
> > **Note:**
> >
> > - This configuration is only valid when the dir and spill-dir of the Raft Engine are specified as different disk drives.
> > - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV:
> >     - Shut down TiKV;
> >     - Copy all raft Logs in this directory to the directory;
> >     - Remove this configuration from the TiKV configuration file;
> >     - Restart TiKV;

Reviewer 3: ```suggestion
+ The auxiliary directory for storing Raft log files. When the disk specified by `dir` is full, new Raft logs will be stored under the directory specified by `spill-dir`. If this directory does not exist after configuration, it will be created when TiKV is started.
+ If this configuration is not set, it means that the directory does not exist.

> **Note:**
>
> - This configuration is only valid when the `dir` and `spill-dir` of the Raft Engine are specified as different disk drives.
> - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV:
>     1. Shut down TiKV.
>     2. Copy all the Raft logs in the `spill-dir` directory to the [`dir`](/tikv-configuration-file.md#dir) directory.
>     3. Remove this configuration from the TiKV configuration file.
>     4. Restart TiKV.
```

> > - This configuration is only valid when the `dir` and `spill-dir` of the Raft Engine are specified as different disk drives.
> > - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV. Otherwise, TiKV cannot start.
> >     1. Shut down TiKV.
> >     2. Copy all the Raft Logs in the `spill-dir` directory to the [`dir`](/tikv-configuration-file.md#dir) directory.

Reviewer: ```suggestion
> - This configuration only takes effect when the `dir` and `spill-dir` of the Raft Engine are set to different disks.
> - After enabling this feature, if you want to disable it, you need to perform the following operations before restarting TiKV. Otherwise, TiKV will fail to start.
>     1. Stop TiKV.
>     2. Copy all the Raft Logs from the `spill-dir` directory to the [`dir`](/tikv-configuration-file.md#dir) directory.
```
